source/common/http/http2/codec_impl.cc

```diff
@@ -1098,6 +1074,99 @@ enum GoAwayErrorCode ngHttp2ErrorCodeToErrorCode(uint32_t code) noexcept {
   }
 }
 
+Status ConnectionImpl::onPing(uint64_t opaque_data, bool is_ack) {
+  if (is_ack) {
+    ENVOY_CONN_LOG(trace, "recv PING ACK {}", connection_, opaque_data);
+
+    onKeepaliveResponse();
+  }
+  return okStatus();
+}
+
+Status ConnectionImpl::onBeginData(int32_t stream_id, size_t length, uint8_t type, uint8_t flags,
+                                   size_t padding) {
+  RETURN_IF_ERROR(trackInboundFrames(stream_id, length, type, flags, padding));
+
+  StreamImpl* stream = getStreamUnchecked(stream_id);
+  if (!stream) {
+    return okStatus();
+  }
+
+  // Track bytes received.
+  stream->bytes_meter_->addWireBytesReceived(length + H2_FRAME_HEADER_SIZE);
+
+  stream->remote_end_stream_ = flags & NGHTTP2_FLAG_END_STREAM;
+  stream->decodeData();
+  return okStatus();
+}
+
+Status ConnectionImpl::onGoAway(uint32_t error_code) {
+  if (!raised_goaway_) {
+    raised_goaway_ = true;
+    callbacks().onGoAway(ngHttp2ErrorCodeToErrorCode(error_code));
+  }
+  return okStatus();
+}
+
+Status ConnectionImpl::onHeaders(int32_t stream_id, size_t length, uint8_t flags,
+                                 int headers_category) {
+  StreamImpl* stream = getStreamUnchecked(stream_id);
```

Comment:
here and in onRstStream(): Could pass a reference to the stream as an argument, saving a lookup.

edit: Maybe another option would be removing the getStreamUnchecked() call in onFrameReceived().

Reply:
Good point, I think we can remove the getStreamUnchecked() call from onFrameReceived().